python/GafferUI/NodeSetEditor.py

```diff
@@ -61,22 +77,149 @@ def __init__( self, topLevelWidget, scriptNode, **kw ) :
 		# to avoid being called when they're only half constructed.
 		self.__setNodeSetInternal( self.scriptNode().selection(), callUpdateFromSet=False )
 
-	## Sets the nodes that will be displayed by this editor. As members are
-	# added to and removed from the set, the UI will be updated automatically
-	# to show them. This also calls `nodeSet.setRemoveOrphans( True )` so that
-	# deleted nodes are not visible in the UI.
+	## Sets the nodes that will be displayed by this editor.
+	# As members are added to and removed from the set, the UI will be updated
+	# automatically to show them. This also calls `nodeSet.setRemoveOrphans( True
+	# )` so that deleted nodes are not visible in the UI.
+	#
+	# This will break any editor links, if set.
+	# NOTE: The driver will be updated *after* the node set, such that calling
```

Comment:
Is there a good reason for this? It seems unexpected - as per my comment for the SourceSet signals, we want to emit signals only when we're in a consistent state from an observer's perspective.

Reply:
Reflected our chat off-line in https://github.com/GafferHQ/gaffer/pull/3270/commits/318060f583e5e0e266c3d592c86800da5bc25fc0